modules/machine-config-node-drain.adoc

```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="machine-config-node-drain_{context}"]
+= Understanding Machine Config Operator node drain behavior 
+
+When you use a machine config to change a system feature, the Machine Config Operator (MCO) applies those changes and ensures that each node is in the desired configuration state.
+
+In the majority of cases, before applying the new rendered-machineconfig, the operator performs the following steps on each node until all the nodes have the updated configuration:
+
+. Cordon. The MCO marks the node as not schedulable for additional workloads.
+. Drain. The MCO terminates all running workloads on the node, causing them to be rescheduled onto other nodes.
+. Apply. The MCO writes any new config files, enables systemd units, sets kernel arguments, or upgrades to a new operating system.
+. Reboot. The MCO restarts the node with the new configuration.
+. Uncordon. The MCO marks the node as schedulable for workloads.
+
+Throughout this process, the MCO ensures that the `MaxUnavailable` value set in the machine config pool is respected.
+
+However, in certain cases the nodes are not drained. For more information, see "Machine config overview."
+
+[NOTE]
+====
+When the MCO drains pods on the master node, note the following conditions:
+
+* The MCO does not drain control plane pods, which have the critical pod annotation. 
+
+* In {sno} clusters, after the control plane is rebooted, the MCO is responsible for restarting the control plane.
```

Comment:
I don't think MCO is responsible to restart control plane. MCO mainly reboots node and rest of the thing is upto other services to come up in their order. MCO doesn't do any additional magic on SNO clusters. 
But yeah, we can mention that on SNO clusters, MCO always skips drain operation.